pallets/identity/src/types.rs

```diff
@@ -31,3 +32,15 @@ pub enum DidStatus {
     Exists,
     CddVerified,
 }
+
+/// Aggregate information about an `AccountId` in relation to an `IdentityId`.
+#[derive(Eq, PartialEq, Encode, Decode)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+pub struct KeyIdData<IdentityId> {
```

Comment:
This record is not based on current requirements. Only secondary keys are given permissions. All primary keys have full permissions. You can use runtime storage `DidRecords` in `identity` to determine whether a key is primary or secondary, and to find permissions of secondary keys.

I suggest collapsing the last two fields in one, possibly of type
```rust
pub enum PrimaryOrSecondaryKeyData {
    Primary,
    Secondary(Permissions),
}
```

Reply:
Collapsed them into `permissions: Option<Permissions>`.